gems/sorbet-runtime/lib/types/private/methods/signature.rb

```diff
@@ -146,29 +146,30 @@ def each_args_value_type(args)
     # can't) match the definition of the method we're validating. In addition, Ruby has a bug that
     # causes forwarding **kwargs to do the wrong thing: see https://bugs.ruby-lang.org/issues/10708
     # and https://bugs.ruby-lang.org/issues/11860.
-    if (args.length > @req_arg_count) && (!@kwarg_types.empty? || @has_keyrest) && args[-1].is_a?(Hash)
+    args_length = args.length
+    if (args_length > @req_arg_count) && (!@kwarg_types.empty? || @has_keyrest) && args[-1].is_a?(Hash)
       kwargs = args[-1]
-      args = args[0...-1]
+      args_length -= 1
```

Comment:
I'm worried about someone accidentally modifying this code in the future. What if we used `Array#pop` to remove the element from the back of the array and stash it in `kwargs`?

Also, it might be simpler if you stash `args_length` once after the `if/else` here, rather than initializing it outside and modifying it in one branch.

Reply:
I'm a little bit afraid of mutating the input. That feels riskier than someone modifying the code in the same function.

Reply:
Ah i didn't see that `args` was an input.